File: app/src/debug/java/com/hana053/micropost/Application.java
package com.hana053.micropost;

import android.os.StrictMode;

import com.squareup.leakcanary.LeakCanary;

import timber.log.Timber;

public class Application extends BaseApplication {

    @Override
    public void onCreate() {
        super.onCreate();
        LeakCanary.install(this);

        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder().detectAll().penaltyLog().penaltyDeathOnNetwork().build());
        StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder().detectLeakedSqlLiteObjects().detectLeakedClosableObjects().detectActivityLeaks().penaltyLog().build());

        Timber.plant(new Timber.DebugTree());
    }

}

File: app/src/debug/java/com/hana053/micropost/Application.kt
package com.hana053.micropost

import android.os.StrictMode
import com.squareup.leakcanary.LeakCanary
import timber.log.Timber

class Application : BaseApplication() {

    override fun onCreate() {
        super.onCreate()
        LeakCanary.install(this)

        StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder().detectAll().penaltyLog().penaltyDeathOnNetwork().build())
        StrictMode.setVmPolicy(StrictMode.VmPolicy.Builder().detectLeakedSqlLiteObjects().detectLeakedClosableObjects().detectActivityLeaks().penaltyLog().build())

        Timber.plant(Timber.DebugTree())
    }

}
